bluemira/radiation_transport/radiation_profile.py

```diff
@@ -0,0 +1,2173 @@
+            b_tor_sep_imp = self.eq.Bt(self.x_sep_imp)
+            self.b_tot_sep_imp = np.hypot(self.b_pol_sep_imp, b_tor_sep_imp)
+
+    def x_point_radiation_z_ext(self, main_ext=None, pfr_ext=0.3, low_div=True):
+        """
+        Simple definition of a radiation region around the x-point.
+        The region is supposed to extend from an arbitrary z coordinate on the
+        main plasma side, to an arbitrary z coordinate on the private flux region side.
+
+        Parameters
+        ----------
+        main_ext: float [m]
+            region extension on the main plasma side
+        pfr_ext: float [m]
+            region extension on the private flux region side
+        low_div: boolean
+            default=True for the lower divertor. If False, upper divertor
+
+        Returns
+        -------
+        z_main: float [m]
+            vertical (z coordinate) extension of the radiation region
+            toward the main plasma
+        z_pfr: float [m]
+            vertical (z coordinate) extension of the radiation region
+            toward the pfr
+        """
+        if main_ext is None:
+            main_ext = abs(self.points["x_point"]["z_low"])
+
+        if low_div is True:
+            z_main = self.points["x_point"]["z_low"] + main_ext
+            z_pfr = self.points["x_point"]["z_low"] - pfr_ext
+        else:
+            z_main = self.points["x_point"]["z_up"] - main_ext
+            z_pfr = self.points["x_point"]["z_up"] + pfr_ext
+
+        return z_main, z_pfr
+
+    def radiation_region_ends(self, z_main, z_pfr, lfs=True):
+        """
+        Entering and exiting points (x, z) of the radiation region
+        detected on the separatrix.
+        The separatrix is supposed to be given by relevant half.
+
+        Parameters
+        ----------
+        z_main: float [m]
+            vertical (z coordinate) extension of the radiation region
+            toward the main plasma
+        z_pfr: float [m]
+            vertical (z coordinate) extension of the radiation region
+            toward the pfr
+        lfs: boolean
+            default=True for the low field side (right half).
+            If False, high field side (left half).
+
+        Returns
+        -------
+        entrance_x, entrance_z: float, float
+            x, z coordinates of the radiation region starting point
+        exit_x, exit_z: float, float
+            x, z coordinates of the radiation region ending point
+        """
+        if self.eq.is_double_null:
+            sep_loop = self.sep_lfs if lfs else self.sep_hfs
+        else:
+            sep_loop = self.sep_ob if lfs else self.sep_ib
+        if z_main > z_pfr:
+            reg_i = np.where((sep_loop.z < z_main) & (sep_loop.z > z_pfr))[0]
+            i_in = np.where(sep_loop.z == np.max(sep_loop.z[reg_i]))[0]
+            i_out = np.where(sep_loop.z == np.min(sep_loop.z[reg_i]))[0]
+        else:
+            reg_i = np.where((sep_loop.z > z_main) & (sep_loop.z < z_pfr))[0]
```

Comment:
should these have a >= or <= and the else includes `z_main <= z_pfr` not just `z_main < z_pfr`?

Reply:
@je-cook corretcted. I had just assumed that it was unlike to get exactly the same value of `z_main` or `z_pfr`

Reply:
yeah definitely unlikely but could lead to a weird bug